tests/cunit/test_iosystem2.c

```diff
@@ -202,6 +202,8 @@ int main(int argc, char **argv)
                 ERR(ret);
             if ((ret = PIOc_closefile(ncid2)))
                 ERR(ret);
+
+            MPI_Barrier(test_comm);
```

Comment:
I have added this to the test, but also checking the return code as I do for all MPI function calls.